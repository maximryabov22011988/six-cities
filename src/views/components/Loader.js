import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import nameSpace from '../../state/name-spaces';

const propTypes = {
  isLoading: PropTypes.bool,
};

const wrapStyles = {
  position: 'fixed',
  left: 0,
  right: 0,
  top: 0,
  bottom: 0,
  zIndex: 10000,
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: 'rgba(255,255,255, 0.85)',
};

class Loader extends React.Component {
  render() {
    const { isLoading } = this.props;
    return isLoading ? (
      <div style={wrapStyles}>
        <div className="lds-ellipsis">
          <div />
          <div />
          <div />
          <div />
        </div>
      </div>
    ) : null;
  }
}

Loader.propTypes = propTypes;

const mapStateToProps = (state) => ({
  isLoading: state[nameSpace.APP].isLoading,
});

export default connect(mapStateToProps)(Loader);
