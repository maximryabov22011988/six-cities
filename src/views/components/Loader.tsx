import * as React from 'react';
import { connect } from 'react-redux';

import nameSpace from '../../state/name-spaces';

interface Props {
  isLoading: boolean;
}

const wrapStyles = {
  position: 'fixed' as 'fixed',
  left: 0,
  right: 0,
  top: 0,
  bottom: 0,
  zIndex: 10000,
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: 'rgba(255,255,255, 0.85)',
};

class Loader extends React.Component<Props> {
  render() {
    const { isLoading } = this.props;
    return isLoading ? (
      <div style={wrapStyles}>
        <div className="lds-ellipsis">
          <div />
          <div />
          <div />
          <div />
        </div>
      </div>
    ) : null;
  }
}

const mapStateToProps = (state) => ({
  isLoading: state[nameSpace.APP].isLoading,
});

export default connect(mapStateToProps)(Loader);
