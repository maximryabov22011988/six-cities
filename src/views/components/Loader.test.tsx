import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { Loader } from './Loader';

it('Loader renders correctly with mandatory props', () => {
  const tree = renderer
    .create(
      <Loader isLoading={true}/>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
