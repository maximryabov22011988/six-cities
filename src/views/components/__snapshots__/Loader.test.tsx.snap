// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders correctly with mandatory props 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(255,255,255, 0.85)",
      "bottom": 0,
      "display": "flex",
      "justifyContent": "center",
      "left": 0,
      "position": "fixed",
      "right": 0,
      "top": 0,
      "zIndex": 10000,
    }
  }
>
  <div
    className="lds-ellipsis"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
